components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -133,6 +133,13 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
     return ContentFilteringPolicy::kAllow;
   }
 
+  // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+  // Doing so can break logins or other site functionality. 
+  if (!IsThirdParty(url, first_party_origin)) {
+    // Downgrade from kBlockRequests to kBlockCookies to minimize impact on sites. 
+    return ContentFilteringPolicy::kBlockCookies;
```

Comment:
Wait, this is going to break stuff. We only want to apply this to URLs that are blocked by the filter. This is not running the filter at all.